test/extended/cluster/cl.go

```diff
@@ -202,6 +208,19 @@ func newProject(nsName string) *projectapi.Project {
 	}
 }
 
+func labelNamespace(c kclientset.Interface, ns string, pairs map[string]string) (*v1.Namespace, error) {
+	if len(pairs) == 0 {
+		return nil, nil
+	}
+	var d struct{ Metadata metav1.ObjectMeta `json:"metadata"`}
```

Comment:
I've never seen a nested struct created like this, but I think the underlying issue is that you're trying to take existing golang types and then convert them to json which is unnecessary. The `Namespace.Labels` are already of type `map[string]string` which is the same type you've added to the `ClusterLoaderType`.